msb/zmq_base/Publisher.py

```diff
@@ -1,36 +1,38 @@
 import zmq
 import pickle
 import sys
+import json
 
 class Publisher:
-    def __init__(self, protocol="tcp", ip="*", port="5555", connect_to = None):
-        if not connect_to:
-            connect_to = f"{protocol}://{ip}:{port}"
-            print(f"binding to : {connect_to}")
+    def __init__(self, config):
```

Comment:
With this implementation, every service, that instantiates an object of type publisher, must also provide a valid publisher / zmq configuration as an argument to the constructor. I'd much prefer it, if the zmq configuration is created implicitly in the constructor of the publisher (or subscriber) class, like so:

```python
from msb.config.parse import read_yaml_config_file, update_config

class Publisher:
    def __init__(self):
        self.config = ZMQConf()   # instantiate a default configuration
        update_config(self.config, read_yaml_config_file("zmq.yaml")) # update configuration from config file file exists
```